### PR TITLE
web-ui: Added Unit Tests for server

### PR DESCRIPTION
**Pull request by bhaskarmelkani**
Added unit tests for server with coverage report.

Related to #179.

### CODE REVIEW

File: web-ui/test/unit/server/create-http-server.spec.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+          createServer: mockCreateHttpServer
+        }
+      });
+  jest.mock('https', () => {
+        return {
+          createServer: mockCreateHttpsServer
+        }
+      });
+  jest.mock('fs', () => {
+        return {
```
Comment (mfellner) [inline web-ui/test/unit/server/create-http-server.spec.js]: These statements are indented a bit strangely. Let's write it like this:

```javascript
jest.mock('fs', () => ({
  readFileSync: mockReadFileSync
});
```

File: web-ui/test/unit/server/create-http-server.spec.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      CreateHttpServer = require('../../../src/server/create-http-server');
+      server = new CreateHttpServer();
+    });
+
+    test('should create a http server', () => {
+      expect(mockCreateHttpServer).toHaveBeenCalled();
+    });
+  });
+
+  describe('when env.SSL_ENABLED is false', function (){
```
Comment (mfellner) [inline web-ui/test/unit/server/create-http-server.spec.js]: Let's use arrow functions `() => { }` consistently everywhere!

File: web-ui/test/unit/server/create-http-server.spec.js (hunk 1 above)
Comment (bhaskarmelkani) [inline web-ui/test/unit/server/create-http-server.spec.js]: 👍

File: web-ui/test/unit/server/create-http-server.spec.js (hunk 2 above)
Comment (bhaskarmelkani) [inline web-ui/test/unit/server/create-http-server.spec.js]: 👍

**Comment by bhaskarmelkani** [comment]
👍

File: web-ui/.gitignore [hunk 3]
```diff
@@ -7,3 +7,4 @@ cert
 credentials
 scm-source.json
 target
+coverage
```
Comment (rbarilani) [inline web-ui/.gitignore]: @bhaskarmelkani the coverage report output directory can be customized with [coverage directory option](https://facebook.github.io/jest/docs/configuration.html#coveragedirectory-string) so you can specify the one that we already use for the other reports (checkstyle): `target`.

File: web-ui/package.json [hunk 4]
```diff
@@ -50,9 +51,19 @@
     "babel-preset-es2015": "6.24.0",
     "babel-preset-react": "6.23.0",
     "eslint": "3.18.0",
+    "jest": "19.0.2",
     "node-scm-source": "0.2.0",
     "nodemon": "1.11.0",
     "webpack": "2.2.1",
     "webpack-dev-server": "2.4.2"
+  },
+  "jest": {
```
Comment (bhaskarmelkani) [inline web-ui/package.json]: Done 👍

File: web-ui/.gitignore (hunk 3 above)
Comment (bhaskarmelkani) [inline web-ui/.gitignore]: Fixed 👍

File: web-ui/test/unit/server/env-handler.spec.js [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+  let envHandler;
+  const mockWrite = jest.fn();
+  const res = {
+    setHeader: () => {},
+    write: mockWrite,
+    end: () => {}
+  };
+  const req = {};
+  const mockReadFile = jest.fn()
+                  .mockImplementationOnce((path, cb) => {
```
Comment (bhaskarmelkani) [inline web-ui/test/unit/server/env-handler.spec.js]: fixed 👍

**Comment by bhaskarmelkani** [comment]
@rbarilani  I will check the best way to have a suite. This setup should work as it is for client as well.
Have added coverage report for all the code(covered/ not covered).

**Comment by bhaskarmelkani** [comment]
👍

**Comment by mfellner** [comment]
👍

**Pull request merged**